hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3579,12 +3579,47 @@ func reconcileKASNetworkPolicy(policy *networkingv1.NetworkPolicy, hcluster *hyp
 		})
 	}
 
+	dnsUDPPort := intstr.FromInt(5353)
+	dnsUDPProtocol := corev1.ProtocolUDP
+	dnsTCPPort := intstr.FromInt(5353)
+	dnsTCPProtocol := corev1.ProtocolTCP
+	policy.Spec.Egress = []networkingv1.NetworkPolicyEgressRule{
+		{
+			To: []networkingv1.NetworkPolicyPeer{
+				{
+					PodSelector: &metav1.LabelSelector{},
+				},
+			},
+		},
+		{
+			To: []networkingv1.NetworkPolicyPeer{
+				{
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"kubernetes.io/metadata.name": "openshift-dns",
+						},
+					},
+				},
+			},
+			Ports: []networkingv1.NetworkPolicyPort{
+				{
+					Port:     &dnsUDPPort,
+					Protocol: &dnsUDPProtocol,
+				},
+				{
+					Port:     &dnsTCPPort,
+					Protocol: &dnsTCPProtocol,
+				},
+			},
+		},
+	}
```

Comment:
do you want to support the kube-virt team in vanilla Kubernetes clusters as well? If so it might be nice to look at the "capabilities" and if not openshift make it so the DNS port is 53

Reply:
actually: looking at our code: we will need that to be valid (we will need to have that support for vanilla Kubernetes as well (it can either be kube-system port 53 or just port 53 in general)

Reply:
and it should be one or the other
(aka when openshift just set the openshift piece), when vanilla kubernetes do the kube-system port 53 piece

Reply:
roger that.  working with RH SD to make sure that just allowing all to port 53 (both UDP/TCP) with no namespace filter works for them.